platform-sdk/swirlds-base/src/main/java/com/swirlds/base/utility/FileSystem.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Provides file system-related utility methods.
+ */
+public final class FileSystem {
+    /**
+     * Private constructor to prevent utility class instantiation.
+     */
+    private FileSystem() {}
+
+    /**
+     * Evaluates a given {@code path} to ensure it exists.
+     *
+     * <p>
+     * This method includes retry logic to handle files which may not yet exist but will become available within a
+     * given threshold. Examples of these cases are Kubernetes projected volumes and persistent volumes.
+     *
+     * <p>
+     * This overloaded method uses a default of 20 attempts with a 2-second delay between each attempt.
+     *
+     * @param path the path to check for existence.
+     * @return true if the file exists before the retries have been exhausted; otherwise false.
+     * @throws NullPointerException if the {@code path} argument is a {@code null} value.
+     */
+    public static boolean waitForPathPresence(@NonNull Path path) {
+        return waitForPathPresence(path, 20, 2_000);
+    }
+
+    /**
+     * Evaluates a given {@code path} to ensure it exists.
+     *
+     * <p>
+     * This method includes retry logic to handle files which may not yet exist but will become available within a
+     * given threshold. Examples of these cases are Kubernetes projected volumes and persistent volumes.
+     *
+     * @param path        the path to check for existence.
+     * @param maxAttempts the maximum number of retry attempts.
+     * @param delayMs     the delay between retry attempts.
+     * @return true if the file exists before the retries have been exhausted; otherwise false.
+     * @throws IllegalArgumentException if the {@code maxAttempts} argument is less than or equal to zero (0) or the
+     *                                 {@code delayMs} argument is less than zero (0).
+     * @throws NullPointerException    if the {@code path} argument is a {@code null} value.
+     */
+    public static boolean waitForPathPresence(@NonNull Path path, int maxAttempts, int delayMs) {
```

Comment:
Rather then using `int maxAttempts, int delayMs`, I suggested just a single parameter `Duration waitTime` with possibly a `Duration retryTime` as well, but I think this second one is not needed, we can just use a sane default. Using an `int` to represent milliseconds can be more confusing